django_project/ta_system/admin.py

```diff
@@ -131,6 +140,20 @@ def assignment_data_download(self, request):
                 )
         return redirect('admin:index')
 
+    def change_system_status(self, request):
+        if request.method != 'POST':
+            return handle_bad_request(request, app='admin', expected_method='POST')
+        if models.SystemStatus.objects.count() > 0:
+            previous_system_status = models.SystemStatus.objects.order_by(
+                'id').last()
+            new_system_status = models.SystemStatus()
```

Comment:
This line can be extracted and placed above, say on line 146, since it appears in both the if and else clause.

Reply:
code removed